crates/db_schema/src/traits.rs

```diff
@@ -197,3 +197,15 @@ pub trait ApubActor {
   where
     Self: Sized;
 }
+
+pub trait InternalToCombinedView {
+  type CombinedView;
+
+  /// Maps the combined DB row to an enum
+  fn map_to_enum(self) -> Option<Self::CombinedView>;
+}
+
+pub trait ToSlimView {
+  type SlimView;
+  fn map_to_slim(self) -> Self::SlimView;
+}
```

Comment:
There is no reason for a trait if its only implemented in one place. Use a normal function instead.

Reply:
It's good to have for the map, and there's a good chance ppl might want other slim views in the future anyway, slim post views (maybe without the body), slim community views (maybe without sidebars), etc.

Reply:
You can do map with a plain function as well, no need for a trait. And traits are mainly useful for generics which are not used here.

Reply:
K I removed the trait. I do like using them with [associated types like this](https://doc.rust-lang.org/rust-by-example/generics/assoc_items/types.html) to define any common behavior, not just for generics.

Reply:
In this case there seems to be no common behaviour with other code.